### PR TITLE
[PLA-8828] Allow working with archived branches.

### PR DESCRIPTION
**Pull request by anoto-moniz**
# Citrine Python PR

## Description 
Adds an archive, restore, and list_archived operation to the BranchCollection.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [X] I have added tests for 100% coverage
- [X] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [X] I have bumped the version in setup.py

### CODE REVIEW

File: tests/resources/test_branch.py [hunk 1]
```diff
@@ -44,18 +44,19 @@ def collection(session) -> BranchCollection:
...
 def branch_path(collection) -> str:
-    return f'/projects/{collection.project_id}/design-workflow-branches'
+    return BranchCollection._path_template.format(project_id=collection.project_id)
 
 
 def test_str(branch):
     assert str(branch) == f'<Branch {branch.name!r}>'
 
 
 def test_branch_build(collection, branch):
```
Comment (talltea) [inline tests/resources/test_branch.py:54]: looks like you don't use `branch` any more in this test

File: tests/resources/test_branch.py [hunk 2]
```diff
@@ -134,7 +135,8 @@ def test_branch_delete(session, collection, branch_path):
 def test_branch_update(session, collection, branch, branch_path):
     # Given
     branch.name = "NEW NAME"
-    session.set_response(branch.dump())
+    branch_data = BranchDataFactory(name=branch.name)
```
Comment (talltea) [inline tests/resources/test_branch.py:138]: probably can just go straight to `branch_data = BranchDataFactory(name="NEW NAME")` without using the `branch` fixture.

File: tests/resources/test_branch.py [hunk 3]
```diff
@@ -94,7 +95,7 @@ def test_branch_register(session, collection, branch_path):
 def test_branch_get(session, collection, branch, branch_path):
```
Comment (talltea) [inline tests/resources/test_branch.py:95]: `branch` isn't used any more. If you ended up replacing all of the fixture uses with the Factory, you can probably 🪓  the fixture itself too. (I didn't check the whole file for uses of it though)

**Comment by talltea** [comment]
This looks great. There are some minor things in the tests, and you'll want to resolve the conflicts, but once that's in, I'll approve.

File: docs/source/workflows/design_workflows.rst [hunk 4]
```diff
@@ -178,3 +178,21 @@ A branch has a name, along with any number of design workflows. A DesignWorkflow
     # status info will contain relevant validation information
     # (i.e. why the workflow is valid/invalid)
     print(validated_workflow.status_info)
+
+
+When you're done with a branch, it can be archived, removing it from the results of ``list`` and sets the ``archived`` flag. ``list_archived`` lists all archived branches on a project. An archived branch can be restore it via its unique ID.
```
Comment (dkao-citrine) [inline docs/source/workflows/design_workflows.rst:183]: Super nitpick: can you break this up into two lines?

File: src/citrine/resources/design_workflow.py [hunk 5]
```diff
@@ -54,7 +54,9 @@ def register(self, model: DesignWorkflow) -> DesignWorkflow:
             # There are a number of contexts in which hitting design workflow endpoints without a
             # branch ID is valid, so only this particular usage is deprecated.
             msg = ('Creating a design workflow without a branch is deprecated as of 1.19.0 and '
-                   'will be removed in 2.0.0.')
+                   'will be removed in 2.0.0. Branches are a concept introduced in the CP2 '
+                   'version of the Citrine Platform. To learn more, see our documentation at '
+                   'https://citrineinformatics.github.io/citrine-python/workflows/design_workflows.html#branches')  # noqa
```
Comment (dkao-citrine) [inline src/citrine/resources/design_workflow.py:59]: Nice

**Comment by dkao-citrine** [comment]
Question for the group: will we need to add an end-to-end test in nextgen-devkit to test this new functionality?

File: tests/resources/test_branch.py (hunk 3 above)
Comment (anoto-moniz) [reply tests/resources/test_branch.py:95]: There were a few other places I used it, but they weren't really necessary, so I ripped it out.

**Review by talltea — APPROVED** [review]

**Review by dkao-citrine — APPROVED** [review]

**Pull request merged**